zoloto/marker.py

```diff
@@ -33,85 +34,101 @@ def id(self) -> int:
     def size(self) -> int:
         return self.__size
 
-    def _is_eager(self) -> bool:
-        return self.__precalculated_vectors is not None
+    @property
+    def marker_dict(self) -> MarkerDict:
+        return self.__marker_dict
 
     @property
     def pixel_corners(self) -> List[Coordinates]:
-        return [Coordinates(*coords) for coords in self.__pixel_corners]
+        return [Coordinates(x=float(x), y=float(y)) for x, y in self._pixel_corners]
 
     @cached_property
     def pixel_centre(self) -> Coordinates:
-        tl, _, br, _ = self.__pixel_corners
-        return Coordinates(
-            x=tl[0] + (self.__size / 2) - 1, y=br[1] - (self.__size / 2),
-        )
+        tl, _, br, _ = self.pixel_corners
+        return Coordinates(x=tl.x + (self.size / 2) - 1, y=br.y - (self.size / 2),)
 
     @cached_property
     def distance(self) -> int:
         return int(linalg.norm(self._tvec))
 
-    @property
+    @cached_property
     def orientation(self) -> Orientation:
         return Orientation(*self._rvec)
 
     @cached_property
     def spherical(self) -> Spherical:
         x, y, z = self._tvec
-        return Spherical(rot_x=arctan2(y, z), rot_y=arctan2(x, z), dist=self.distance)
+        return Spherical(
+            rot_x=float(arctan2(y, z)), rot_y=float(arctan2(x, z)), dist=self.distance
+        )
 
     @property
```

Comment:
Might be good to make this `cached_property` whilst we're refactoring.

Reply:
My intention is to only make things `cached_property`s if it's really needed, for performance reasons.